src/wrappers/lambda.js

```diff
@@ -33,12 +33,12 @@ function baseLambdaWrapper(
 ) {
     // eslint-disable-next-line consistent-return
     return (originalEvent, originalContext, originalCallback) => {
-        tracer.restart();
         let runner;
         let callbackCalled = false;
 
         try {
             runner = lambdaRunner.createRunner(originalContext, runnerResourceType);
```

Comment:
I think this will be problemtic since if we fail in runner creation we will send the previous trace. Maybe we have to split it to restart and addRunner